torch/_inductor/utils.py

```diff
@@ -838,20 +841,34 @@ def stringfy_layout(layout: ir.Layout | None) -> str:
                         if buffer is None:
                             continue
                         input_name, layout = get_buffer_info(buffer, r.name)
+                        
+                        # For split reductions, we need to identify intermediate buffers
+                        # that are outputs of previous kernels in the split sequence
+                        if hasattr(n, 'node') and hasattr(n.node, 'data'):
+                            if hasattr(n.node.data, 'reduction_type') and n.node.data.reduction_type:
+                                # This is a reduction node, check if it's part of a split reduction
+                                if hasattr(n.node.data, 'num_splits') and n.node.data.num_splits > 1:
+                                    split_reduction_buffers.add(input_name)
+                        
                         detailed_metadata.append(
                             f"{wrapper.comment}   %{input_name} : Tensor "
                             f"{stringfy_layout(layout)} = PlaceHolder[target={input_name}]"
                         )
 
                 if (
-                    hasattr(n.read_writes, "writes")
-                    and n.read_writes.writes is not None
+                    hasattr(n.read_writes, "writes") and n.read_writes.writes is not None
                 ):
                     for w in n.read_writes.writes:
                         buffer = V.graph.try_get_buffer(w.name)
                         if buffer is None:
                             continue
                         output_name, _ = get_buffer_info(buffer, w.name)
+                        
+                        # Track split reduction output buffers
+                        if hasattr(n, 'node') and hasattr(n.node, 'data'):
+                            if hasattr(n.node.data, 'reduction_type') and n.node.data.reduction_type:
+                                if hasattr(n.node.data, 'num_splits') and n.node.data.num_splits > 1:
```

Comment:
Put this repeated check in a virtual method.  That might allow removing some of the hasattrs() as well.